great_expectations/profile/user_configurable_profiler.py

```diff
@@ -770,14 +792,31 @@ def _build_expectations_numeric(self, dataset, column, **kwargs):
                 )
 
         # quantile values
+        # if (
+        #     "expect_column_quantile_values_to_be_between"
+        #     not in self.excluded_expectations
+        # ) and isinstance(dataset, Dataset):
         if (
             "expect_column_quantile_values_to_be_between"
             not in self.excluded_expectations
         ):
             if isinstance(dataset, PandasDataset):
-                allow_relative_error = "lower"
-            else:
-                allow_relative_error = dataset.attempt_allowing_relative_error()
+                if isinstance(dataset, PandasDataset):
+                    allow_relative_error = "lower"
+                else:
+                    allow_relative_error = dataset.attempt_allowing_relative_error()
+            elif isinstance(dataset, Validator):
+                if isinstance(dataset.execution_engine, PandasExecutionEngine):
+                    allow_relative_error = "lower"
+                if isinstance(dataset.execution_engine, SparkDFExecutionEngine):
+                    allow_relative_error = 0.0
+                if isinstance(dataset.execution_engine, SqlAlchemyExecutionEngine):
+                    allow_relative_error = (
+                        attempt_allowing_relative_error(
+                            dataset.execution_engine.drivername
+                        )
+                        or True
+                    )
```

Comment:
What's the purpose of the `or True`? Wouldn't that make that expression always evaluate to `True`?

Reply:
Yup - great catch, and it also helped me realize that I was checking dialect against the wrong object here.